services/CollectionService/src/index.php

```diff
@@ -122,12 +122,14 @@
     //lets invoke the controller method directly
     $responsePut = $app['islandora.resourcecontroller']->put($app, $subRequestPut, $responsePost->headers->get('location'), "members");
     if (201 == $responsePut->getStatusCode()) {// OK, indirect container created
-      //Return only the last created resource
+      //Include headers from the parent one, some of the last one. Basically rewrite everything
       $putHeaders = $responsePut->getHeaders();
       //Guzzle psr7 response objects are inmutable. So we have to make this an array and add directly
-      $putHeaders['Link'] = array('<'.$urlRoute.$tmpUuid.'/members>; rel="alternate"');
-
-      return new Response($responsePut->getBody(), 201, $putHeaders);
+      $putHeaders['Link'] = array('<'.$responsePut->getBody().'>; rel="alternate"');
+      $putHeaders['Link'] = array('<'.$urlRoute.$tmpUuid.'/members>; rel="member"');
+      $putHeaders['Location'] = array($urlRoute.$tmpUuid);
+      //Should i care about the etag?
+      return new Response($putHeaders['Location'][0], 201, $putHeaders);
```

Comment:
Not sure, we are returning the "etag" fedora4 gave us for the indirect container, but our response mutated to be a mixed thing of indirect and pcdm collection responses. So it's just a comment because i don't know what the use of stags during resource creation is (we are not caching this, etags make more sense when asking for resources). Anyway, i suppose @daniel-dgi will know better and remove my comments when moving towards a service provider